packages/server/controllers/auth.js

```diff
@@ -189,6 +190,100 @@ async function authenticate(request, response, next) {
   }
 }
 
+async function verifyUserEmail(email) {
+  const result = await prisma.users.findUnique({
+    where: {
+      email: email,
+    },
+  });
+
+  if (result === null) return false;
+
+  return true;
+}
+
+async function isGoogleAccount(userId) {
+  const result = await prisma.users.findUnique({
+    where: {
+      userId: userId,
+    },
+  });
+
+  if (result.password == null) return true;
+
+  return false;
+}
+
+// TODO: implement send password reset email
+async function sendPasswordResetEmail(email, link) {
+  try {
+    const transporter = nodemailer.createTransport({
+      host: "mail.privateemail.com",
+      port: 465,
+      secure: true,
+      auth: {
+        user: "icebreak@cppicebreak.com",
+        pass: process.env.MAILBOX_PASSWORD,
+      },
+    });
+
+    const info = await transporter.sendMail({
+      from: "icebreak@cppicebreak.com", // sender address
+      to: email, // list of receivers
+      subject: "Icebreak: Password Reset Request", // Subject line
+      text: `Please click the following link to reset your password: ${link}`, // plain text body
+      // html: "<b>Hello world?</b>", // html body (THE HTML email can be a frontend task.)
+    });
```

Comment:
If you take a closer look at when you're creating the transport in this `sendPasswordResetEmail(email, link)` function and the `sendPasswordResetConfirmationEmail(email)` function, there is duplicate code where the only things being changed are the email subject and the email text.

Generally, any duplicated code is a sign that you need to rewrite it. Good practice is to reduce duplicate code and make code as reuseable as possible. My suggestion is to write a base `sendEmail(address, subject, body)` function and then change your `sendPasswordResetEmail()` and `sendPasswordResetConfirmationEmail()` to call the new `sendEmail(address, subject, body)`.

Structuring the code this way allows for easy extension to send different emails when writing new features in the future.

```js
async function sendEmail(address, subject, body) {
  /** 
   * no try/catch because possible errors are thrown up
   * until it reaches the route handler try/catch
   */ 
  const transporter = nodemailer.createTransport({
      host: "mail.privateemail.com",
      port: 465,
      secure: true,
      auth: {
        user: "icebreak@cppicebreak.com",
        pass: process.env.MAILBOX_PASSWORD,
      },
    });

    const info = await transporter.sendMail({
      from: "icebreak@cppicebreak.com",
      to: address,
      subject: subject,
      text: body,
    });
}

/**
 * Right now, you guys pass the link to sendPasswordResetEmail
 * but you should instead pass the password reset token and form the link
 * inside the function.
 * 
 * It's better to create and keep the link contained within the function where
 * it is directly used.
 */ 
async function sendPasswordResetEmail(address, passwordResetToken) {
  const subject = "Icebreak: Password Reset Request";
  const link = `http://cppicebreak.com/reset-password?token=${passwordResetToken}`; // changing to icebreak domain for demonstration
  const body = `Please click the following link to reset your password: ${link}`;

  sendEmail(address, subject, body);
}

async function sendPasswordResetConfirmationEmail(address) {
  const subject = "Icebreak: Password Changed";
  const body = "Your Icebreak password has been changed. If you did not request a password change, please immediately contact us at icebreak@cppicebreak.com.";

  sendEmail(address, subject, body);
}
```